trl/trainer/ppo_trainer.py

```diff
@@ -660,12 +663,12 @@ def log_stats(
                 dist.all_reduce(rewards, op=torch.distributed.ReduceOp.SUM)
                 rewards /= self.accelerator.num_processes
 
-            if self.config.log_with_wandb:
+            if self.config.log_with == "wandb":
                 wandb_logs.update(stats)
                 wandb_logs["env/reward_mean"] = torch.mean(rewards).cpu().numpy()
                 wandb_logs["env/reward_std"] = torch.std(rewards).cpu().numpy()
                 wandb_logs["env/reward_dist"] = rewards.cpu().numpy()
-                wandb.log(wandb_logs)
+                self.accelerator.log(wandb_logs)
```

Comment:
since the user can select other frameworks (e.g. tensorboard) we should also log if we are not using wandb. maybe that whole block can be wandb agnostic, what do you think?

Reply:
Sounds good, based on your suggestion I proposed dc8cda2 , here is the corresponding log: https://wandb.ai/distill-bloom/trl/runs/baw84iyv?workspace=user-